types/index.d.ts

```diff
@@ -0,0 +1,116 @@
+import { ObjectSchema, Schema } from 'yup'
+import { ReactElement, ReactNode, SyntheticEvent, ReactType } from 'react'
+import { RenderedAsComponent } from './helpers'
+
+type Errors = Record<string, any[]>
+type Touched = Record<string, boolean>
+
+export type SchemaValue<T extends ObjectSchema<unknown>> = ReturnType<T['cast']>
+
+type ValidateData = {
+  fields: string[]
+  type: SyntheticEvent
+  args?: any[]
+}
+
+type BeforeSubmitData<T> = {
+  value: T
+  errors: Errors
+}
+
+type Mapper<TOut, TIn = any> = (input: TIn) => TOut
+
+type Meta<TValue = any, TSchema extends Schema = Schema<TValue>> = {
```

Comment:
you could do `Meta<TValue = any` and then `schema: Schema<TValue> | undefined`. also you should almost always do `unknown` in these cases, it's safer